bridge/client/app/_components/ktb-project-settings-git-extended/ktb-project-settings-git-extended.component.ts

```diff
@@ -56,7 +46,41 @@ export class KtbProjectSettingsGitExtendedComponent {
   public resetTouched = new EventEmitter<void>();
 
   public get gitData(): IGitDataExtended | undefined {
-    return this.selectedForm === GitFormType.HTTPS ? this.gitDataHttps : this.gitDataSsh;
+    switch (this.selectedForm) {
+      case GitFormType.HTTPS:
+        return this.gitDataHttps;
+      case GitFormType.SSH:
+        return this.gitDataSsh;
+      case GitFormType.NO_UPSTREAM:
+        return { noupstream: '' };
+      default:
+        return undefined;
+    }
+  }
+
+  public ngOnInit(): void {
+    if (!this.gitInputData || this.isRemoteUrlEmpty()) {
+      this.selectedForm = this.required ? GitFormType.HTTPS : GitFormType.NO_UPSTREAM;
+
+      if (this.selectedForm === GitFormType.NO_UPSTREAM) {
+        this.dataChanged(this.selectedForm, this.gitData);
```

Comment:
We now have a problem here I guess. On input, this is emitted which means that there is a change and this means that the popup for leaving the project page will show up. The popup will always show up if you go to the settings of a project and `no upstream` is selected.

Reply:
We could additionally provide a state `UPDATE` and `INIT` to the `IGitDataExtended` and based on that we can decide if the form is touched or not.